x-pack/plugins/security_solution/public/resolver/store/data/reducer.ts

```diff
@@ -36,6 +38,17 @@ export const dataReducer: Reducer<DataState, ResolverAction> = (state = initialS
       ...state,
       hasError: true,
     };
+  } else if (action.type === 'userRequestedRelatedEventData') {
+    return {
+      ...state,
+      relatedEventsReady: new Map(state.relatedEventsReady.set(action.payload, false)),
+    };
+  } else if (action.type === 'serverReturnedRelatedEventData') {
+    return {
+      ...state,
+      relatedEventsReady: new Map(state.relatedEventsReady.set(action.payload.entityID, true)),
```

Comment:
can you refactor this to avoid changing state.

Reply:
https://github.com/elastic/kibana/pull/69098/commits/89bd6a862b9b9b9903726a2130ef2db681fd1068